lib/adapter.js

```diff
@@ -7149,25 +7121,20 @@ function Adapter(options) {
                 Object.keys(this.aliases).forEach(sourceId => {
                     for (let i = this.aliases[sourceId].targets.length - 1; i >= 0; i--) {
                         if (this.aliases[sourceId].targets[i].pattern === aliasPattern) {
-                            count++;
-                            this._removeAliasSubscribe(sourceId, i, () => {
-                                if (!--count) {
-                                    // if no one alias subscribed
-                                    if (!Object.keys(this.aliases).length) {
-                                        if (this._aliasObjectsSubscribed) {
-                                            this._aliasObjectsSubscribed = false;
-                                            adapterObjects.unsubscribe(`${ALIAS_STARTS_WITH}*`);
-                                        }
-                                    }
-                                    typeof callback === 'function' && callback();
-                                }
-                            });
+                            promises.push(new Promise(resolve => this._removeAliasSubscribe(sourceId, i, resolve)));
                         }
                     }
                 });
-            } else {
-                !count && typeof callback === 'function' && callback();
```

Comment:
I'm unsure this else is really replaceabla that way because the logic above might not be executed here

Reply:
empty array will just resolve in promise.all, so yes logic should apply